core/src/com/unciv/logic/trade/TradeEvaluation.kt

```diff
@@ -150,6 +150,19 @@ class TradeEvaluation {
                     ThreatLevel.VeryHigh -> return 1000
                 }
             }
+            TradeType.OfferPeaceTreaty -> {
+                val civToSignPeaceWith = civInfo.gameInfo.getCivilization(offer.name)
+                val threatToThem = Automation.threatAssessment(civToSignPeaceWith, civInfo)
+
+                if (civInfo.isAtWarWith(civToSignPeaceWith)) return 0 // why should we pay you to sign peace...?
+                else when (threatToThem) {
+                    ThreatLevel.VeryLow -> return 0
+                    ThreatLevel.Low -> return 0
+                    ThreatLevel.Medium -> return 200
```

Comment:
Why in general would a civ be interested that we sign peace with someone else? The more of a threat they are, the worse it will be for them - because now they need to worry that the civ we're peaceing with might declare eat on *them*!

Reply:
Good point, I changed it so that it only works with city states that they are allied with if they are threatend.